File: src/main/java/edu/luc/comp433/api/cxf/exception/ApiError.java
package edu.luc.comp433.api.cxf.exception;

import lombok.Getter;

import javax.ws.rs.core.Response;
import java.util.Arrays;
import java.util.List;

@Getter
public class ApiError {

    private Response.Status status;
    private String message;
    private List<String> errors;

    public ApiError(Response.Status status, String message, List<String> errors) {
        super();
        this.status = status;
        this.message = message;
        this.errors = errors;
    }

    public ApiError(Response.Status status, String message, String error) {
        super();
        this.status = status;
        this.message = message;
        errors = Arrays.asList(error);
    }

    public ApiError() {
    }
}

File: front-end/api/ApiError.ts
export default class ApiError extends Error {
    private readonly _status: string;
    private readonly _errors: Array<string>;


    constructor(status: string, message: string, errors: Array<string>) {
        super(message);
        this._status = status;
        this._errors = errors;
    }

    get status(): string {
        return this._status;
    }

    get errors(): Array<string> {
        return this._errors;
    }

}